Suggest matching commands if the user makes a typo

## Proposed API

`frontity creat`

As per now, it warns the user such that the command is not being recognized. It would be great to have a command suggestion logic such that the nearest matching command (`create` in this case) is being suggested.

Also, these errors seem to pop up while compiling the source locally:-

`npm run build`

```bash

node_modules/@emotion/core/types/index.d.ts:82:16 - error TS2665: Invalid module name in augmentation. Module 'react' resolves to an untyped module at '/home/james-the-hacker/CodeSpace/GitHub-Projects/frontity/packages/frontity/node_modules/react/index.js', which cannot be augmented.

82 declare module 'react' {
                  ~~~~~~~

node_modules/@emotion/styled-base/types/helper.d.ts:7:23 - error TS2694: Namespace 'global.JSX' has no exported member 'IntrinsicElements'.

7   C extends keyof JSX.IntrinsicElements | React.JSXElementConstructor<any>
                        ~~~~~~~~~~~~~~~~~

node_modules/@emotion/styled-base/types/helper.d.ts:8:9 - error TS2694: Namespace 'global.JSX' has no exported member 'LibraryManagedAttributes'.

8 > = JSX.LibraryManagedAttributes<C, React.ComponentPropsWithRef<C>>
          ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@emotion/styled-base/types/index.d.ts:29:20 - error TS2694: Namespace 'global.JSX' has no exported member 'IntrinsicElements'.

29 type JSXInEl = JSX.IntrinsicElements
                      ~~~~~~~~~~~~~~~~~

node_modules/@frontity/connect/src/index.d.ts:106:14 - error TS2503: Cannot find namespace 'React'.

106   Component: React.ComponentType<Props>
                 ~~~~~

node_modules/@frontity/connect/src/index.d.ts:109:24 - error TS2503: Cannot find namespace 'React'.

109 export const Provider: React.ProviderExoticComponent<React.ProviderProps<any>>;
                           ~~~~~

node_modules/@frontity/connect/src/index.d.ts:109:54 - error TS2503: Cannot find namespace 'React'.

109 export const Provider: React.ProviderExoticComponent<React.ProviderProps<any>>;
                                                         ~~~~~

node_modules/@frontity/core/dist/types.d.ts:1:34 - error TS2307: Cannot find module 'babel-core'.

1 import { TransformOptions } from "babel-core";
                                   ~~~~~~~~~~~~

node_modules/@frontity/types/src/package.ts:4:26 - error TS2503: Cannot find namespace 'React'.

4     [namespace: string]: React.ReactType;
                           ~~~~~

node_modules/@frontity/types/src/package.ts:7:26 - error TS2503: Cannot find namespace 'React'.

7     [namespace: string]: React.ReactType;
                           ~~~~~


Found 10 errors.

```

--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
     "reinstall": "lerna clean --yes && npm run prepare"
   },
   "devDependencies": {
+    "@types/didyoumean": "^1.2.0",
     "@typescript-eslint/eslint-plugin": "^1.4.2",
     "@typescript-eslint/parser": "^1.4.2",
     "eslint": "^5.14.1",
@@ -23,5 +24,7 @@
     "typescript": "^3.4.5"
   },
   "prettier": {},
-  "dependencies": {}
+  "dependencies": {
+    "didyoumean": "^1.2.1"
+  }
 }

--- a/packages/frontity/src/actions/unknown.ts
+++ b/packages/frontity/src/actions/unknown.ts
@@ -1,7 +1,26 @@
+import didYouMean from "didyoumean";
 import { CommanderStatic } from "commander";
 import chalk from "chalk";
 
+/**
+ * @param {String} command - Unknown command
+ * @param {CommanderStatic} program - commander instance
+ * 
+ * @returns {void}
+ */
+
 export default (command: string, program: CommanderStatic) => {
-  console.log(chalk.red(`Unknown command: ${chalk.bold(command)}\n`));
+  console.log(chalk.red(`Unknown command: ${chalk.bold(command)}`));
+  console.log();
+
+  const availableCommands: string[] = program.commands.map(c => c._name);
+  const suggestion: string | string[] = didYouMean(command, availableCommands);
+  
+  if (suggestion) {
+	console.log(chalk.cyan(`Did you mean '${suggestion}'?`));
+	console.log();
+  }
+  
   program.help();
 };
+

--- a/packages/frontity/src/commands.ts
+++ b/packages/frontity/src/commands.ts
@@ -83,7 +83,7 @@ program
   .description("Get environment information for debugging and issue reporting.")
   .action(info);
 
-program.on("command:*", (command: string) => unknown(command, program));
+program.on("command:*", ([command]: string[]) => unknown(command, program));
 
 // Parses the arguments and adds them to the `command` object.
 program.parse(process.argv);

